### PR TITLE
Adds support for 'supervisor' init_style

### PR DESCRIPTION
**Pull request by logankoester**
I've added support for the [supervisor](http://supervisord.org/) `init_style`.

This is useful for running `consul-template` inside a Docker container alongside the process you want to configure.

### CODE REVIEW

File: attributes/default.rb [hunk 1]
```diff
@@ -2,7 +2,7 @@
 # attributes::default
 #
 
-default['consul_template']['base_url'] = 'https://github.com/hashicorp/consul-template/releases/download/'
+default['consul_template']['base_url'] = 'https://releases.hashicorp.com/consul-template/'
```
Comment (logankoester) [inline attributes/default.rb]: > NOTE! Releases have been moved to https://releases.hashicorp.com/!

See https://github.com/hashicorp/consul-template/releases/

File: recipes/install_binary.rb [hunk 2]
```diff
@@ -27,7 +27,7 @@
   version node['consul_template']['version']
   checksum install_checksum
   url ::URI.join(node['consul_template']['base_url'],
-                 "v#{node['consul_template']['version']}/",
+                 "#{node['consul_template']['version']}/",
```
Comment (logankoester) [inline recipes/install_binary.rb]: URLs have dropped the 'v' since the move, eg https://releases.hashicorp.com/consul-template/0.12.0/

**Comment by adamkrone** [comment]
@logankoester This looks great, and thanks for adding tests!

Can you pull in the latest changes from master and resolve the merge conflict? I just merged another PR that included changes for the new releases.hashicorp.com format, which probably caused it.

Thanks!

**Comment by adamkrone** [comment]
@logankoester I'm going to be releasing a new version of this cookbook soon, and I'm interested in including this. Unfortunately I can't merge this PR until the merge conflicts have been resolved.

If you don't have the time to look into this, let me know and I can try to merge/resolve this locally.

**Pull request merged**